libs/@local/hash-validation/src/data_type.rs

```diff
@@ -786,377 +1134,375 @@ mod tests {
-            "2023-12-22T13:58:26.9+0100",       // %Y-%M-%DT%h:%m:%.1s%Z%z
-            "2023-12-22T13:58:26.95+0100",      // %Y-%M-%DT%h:%m:%.2s%Z%z
-            "2023-12-22T13:58:26.950+0100",     // %Y-%M-%DT%h:%m:%.3s%Z%z
-            "2023-12-22 13:58+0100",            // %Y-%M-%D %h:%m%Z%z
-            "2023-12-22 13:58:26+0100",         // %Y-%M-%D %h:%m:%s%Z%z
-            "2023-12-22 13:58:26.9+0100",       // %Y-%M-%D %h:%m:%.1s%Z%z
-            "2023-12-22 13:58:26.95+0100",      // %Y-%M-%D %h:%m:%.2s%Z%z
-            "2023-12-22 13:58:26.950+0100",     // %Y-%M-%D %h:%m:%.3s%Z%z
-            "2023-12-22T21:43:26+0845",         // %Y-%M-%DT%h:%m:%s+0845
-            "2023-12-22T12:58:26+0000",         // %Y-%M-%DT%h:%m:%s+0000
-            "2023-12-22T13:00:58.950+0000",     // %Y-%M-%DT%h:%m:%.3s+0000
+            "2023-12-22t17:48:15z",             // %Y-%M-%Dt%h:%m:%sz
+            "2023-12-22t17:48:15.083z",         // %Y-%M-%Dt%h:%m:%.3sz
+            "2023-12-22 18:48:15+01:00",        // %Y-%M-%D %h:%m:%s%Z:%z
+            "2023-12-22 18:48:15.0+01:00",      // %Y-%M-%D %h:%m:%.1s%Z:%z
+            "2023-12-22 18:48:15.08+01:00",     // %Y-%M-%D %h:%m:%.2s%Z:%z
+            "2023-12-22 18:48:15.083+01:00",    // %Y-%M-%D %h:%m:%.3s%Z:%z
+            "2023-12-22 18:48:15.083212+01:00", // %Y-%M-%D %h:%m:%s.%u%Z:%z
+            "2023-12-22 17:48:15Z",             // %Y-%M-%D %h:%m:%sZ
+            "2023-12-22 17:48:15z",             // %Y-%M-%D %h:%m:%sz
+            "2023-12-22 17:48:15.0Z",           // %Y-%M-%D %h:%m:%.1sZ
+            "2023-12-22 17:48:15.08Z",          // %Y-%M-%D %h:%m:%.2sZ
+            "2023-12-22 17:48:15.083Z",         // %Y-%M-%D %h:%m:%.3sZ
+            "2023-12-22 17:48:15.083212Z",      // %Y-%M-%D %h:%m:%s.%uZ
+            "2023-12-22 17:48:15.083z",         // %Y-%M-%D %h:%m:%.3sz
+            "2023-12-22 17:48:15.083212z",      // %Y-%M-%D %h:%m:%s.%uz
+            "2023-12-22 17:48:15-00:00",        // %Y-%M-%D %h:%m:%s-00:00
+            "2023-12-22 17:48:15.083-00:00",    // %Y-%M-%D %h:%m:%.3s-00:00
+            "2023-12-22_17:48:15Z",             // %Y-%M-%D_%h:%m:%sZ
+            "2023-12-22_17:48:15z",             // %Y-%M-%D_%h:%m:%sz
+            "2023-12-22_17:48:15.083Z",         // %Y-%M-%D_%h:%m:%.3sZ
+            "2023-12-22_17:48:15.083212Z",      // %Y-%M-%D_%h:%m:%s.%uZ
+            "2023-12-22_17:48:15.083z",         // %Y-%M-%D_%h:%m:%.3sz
+            "2023-12-22_17:48:15.083212z",      // %Y-%M-%D_%h:%m:%s.%uz
+            "2023-12-22T18",                    // %Y-%M-%DT%h
+            "2023-12-22T18,8",                  // %Y-%M-%DT%,1h
+            "2023-12-22T18.8",                  // %Y-%M-%DT%.1h
+            "2023-12-22T18:48",                 // %Y-%M-%DT%h:%m
+            "2023-12-22T18:48,2",               // %Y-%M-%DT%h:%,1m
+            "2023-12-22T18:48.2",               // %Y-%M-%DT%h:%.1m
+            "2023-12-22T18:48:15",              // %Y-%M-%DT%h:%m:%s
```

Comment:
This is the format that the HTML `datetime-local` input outputs (the only native browser datetime picker), so we need to decide what to do to make it into a string with timezone, otherwise the value generated by the input will not validate.

Two options:
1. Whatever datetime the user picked, we store it as UTC (no conversion), i.e. just add a `Z`
    - pros: simple. closest to making no assumptions about what the user intended (although still involves an assumption that they _didn't_ want their timezone preserved)
    - cons: we don't know what timezone the user was in when they picked the datetime
2. We calculate the UTC offset from whatever timezone is set in their browser/device and store that
    - pros: we preserve the user's timezone (at least according to the browser)
    - cons: maybe they didn't want this

In both scenarios we could turn a 'with timezone' back into 'without timezone' if we wanted.

We will need to either 
1. make a decision about this and update the frontend to implement it before merging the PR, or
2. allow timestamps without timezones

We should also update the description of the `DateTime` data type to reflect the decision.

cc @nonparibus as it's probably worth discussing

Reply:
I would probably prefer to add two data types, one with timezone, one without timezone. Mainly, because none of the three solution is satisfying.

Reply:
We discucssed and agreed to do (2) for now, but to add an explicit 'DateTime without TZ' later

Reply:
I will do (2) in the next ~day, we can merge this for now since we won't be attempting to input datetimes via the frontend in the meantime

Reply:
To avoid confusion in the future do we want to explicitly mention in the description that the date time is with a timezone in the `Date Time` data type"? Something like
> A date and time with time zone in the format of RFC 3336.

Reply:
Yes, I think we should do that. Can probably update the seed script and then amend in db directly to avoid a migration (as no-one else has run the data type migration, AFAIK)

Reply:
Sounds good!